apiserver/facades/client/application/deployrepository.go

```diff
@@ -0,0 +1,82 @@
+// Copyright 2023 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package application
+
+import (
+	"github.com/juju/errors"
+	"github.com/juju/featureflag"
+
+	apiservererrors "github.com/juju/juju/apiserver/errors"
+	"github.com/juju/juju/feature"
+	"github.com/juju/juju/rpc/params"
+)
+
+// DeployFromRepository is a one-stop deployment method for repository
+// charms. Only a charm name is required to deploy. If argument validation
+// fails, a list of all errors found in validation will be returned. If a
+// local resource is provided, details required for uploading the validated
+// resource will be returned.
+func (api *APIBase) DeployFromRepository(args params.DeployFromRepositoryArgs) (params.DeployFromRepositoryResults, error) {
```

Comment:
Maybe a comment mentioning that this is available since version 18?

Reply:
Per conversations, we will discuss changing this across the juju code base. No change here yet.